lib/std/Build/Step/Compile.zig

```diff
@@ -433,7 +433,7 @@ pub fn create(owner: *std.Build, options: Options) *Compile {
 
     const self = owner.allocator.create(Compile) catch @panic("OOM");
     self.* = Compile{
-        .strip = null,
+        .strip = if (options.optimize == .ReleaseSmall) true else null,
```

Comment:
The previous value was correct. The default is decided by the compiler rather than the build system. `null` means to use the compiler's default.